### PR TITLE
New Zalando Rule: Must Provide API Audience

### LINKED ISSUE #669: New Check for API Audience
A new check for the Zalando rule [219 - Must provide API Audience](http://zalando.github.io/restful-api-guidelines/#219) has to be implemented.

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Closes #669

### CODE REVIEW

**Comment by maxim-tschumak** [comment]
:-1:

**Comment by maxim-tschumak** [comment]
:+1:

File: server/src/main/java/de/zalando/zally/rule/zalando/ApiAudienceRule.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    @Check(severity = Severity.MUST)
+    fun validate(swagger: Swagger): Violation? {
+        val audience = swagger.info?.vendorExtensions?.get(extensionName)
+
+        return if (audience == null || audience !is String) {
+            Violation(noApiAudienceDesc, listOf(path))
+        } else {
+            if (validAudiences.contains(audience)) null else Violation(invalidApiAudienceDesc, listOf(path))
+        }
```
Comment (roxspring) [inline server/src/main/java/de/zalando/zally/rule/zalando/ApiAudienceRule.kt]: Would likely be a touch clearer using a Kotlin when expression.

Comment (maxim-tschumak) [reply server/src/main/java/de/zalando/zally/rule/zalando/ApiAudienceRule.kt]: Thanks for the hint, didn't know about `when` expressions

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by mfellner** [comment]
👍

**Comment by maxim-tschumak** [comment]
:+1:

**Comment by mfellner** [comment]
👍

**Pull request merged**